### ISSUE TITLE
CA2254 should not apply when using nameof

### ISSUE THREAD

**Issue by datvm** [dotnet/roslyn-analyzers#7606]
### Analyzer

**Diagnostic ID**: [CA2254](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2254): `Template should be a static expression`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 9.0.201

### Describe the bug

When writing a log error with `nameof` like this, `CA2254` is raised

```
logger.LogError($"Failed to find {nameof(Foo)} with id {id}.", id);
```

### Steps To Reproduce

See above

### Expected behavior

The above string should not produce `CA2254` because it's actually static upon compilation.

### Actual behavior

`CA2254` is raised.

**Comment by datvm** [comment]
Sorry it was my mistake, the `$""` causes `{id}` to becomes non-static.
